src/unix/udp.c

```diff
@@ -532,6 +533,12 @@ int uv__udp_bind(uv_udp_t* handle,
       return err;
   }
 
+  if (flags & UV_UDP_REUSEPORT) {
```

Comment:
What is the effect of setting both reuseaddr and reuseport? Is setting the latter enough for the load sharing case? Let's have that documented please.

Reply:
For platforms where `SO_REUSEPORT`s have load balancing, setting both options is allowed and `SO_REUSEPORT` will always override the behavior of `SO_REUSEADDR`.

For platforms where `SO_REUSEPORT`s don't have load balancing, setting both options will fail: `uv__set_reuse()` will succeed but `uv__sock_reuseport` will return `UV_ENOTSUP`.

Reply:
Done. Added documentations along with new tests.